### DISCUSSION TITLE
A TrueNAS script installation ?

### DISCUSSION THREAD

**Discussion by OverStyleFR** [tteck/Proxmox#2956] [Ideas]
Hi,

What about a TrueNAS (Scale and Core) installation script ? You make already one for Nextcloud, OMV and that kind of stuff, why not one for TrueNAS ?

**Comment by halpcomputar** [comment]
I'm sure anyone that's clever enough to pull this off would already be aware, but just in case: TrueNAS can be particularly tricky if not on baremetal to begin with - ie even in a VM it's not ideal.

Best practice for TrueNAS in a VM, though is that, generally rather than passing through individual disks to the VM, you have all disks that will be used in TrueNAS' pool(s) connected to an HBA, and then you pass the entire HBA through to the VM, then create your ZFS pool from within TrueNAS. I would assume the same would be true in the case of trying to containerize TrueNAS.

I would also assume TrueNAS Core wouldn't be possible with its BSD base, but I could be wrong.

**Comment by gh2023-aesun** [comment]
Second this, as it is tricky and a script to handle the correct way to do this would be great.

**Comment by Mellowlynx** [comment]
Just like a firewall, an NAS (or TrueNAS in this case) should always be on bare metal. Don't overcomplicate things that don't need to be.
